File: VotingSystem.React/votingsystem.react.client/src/api/models/LoginRequestDto.ts
import * as yup from "yup";

export interface LoginRequestDto {
    email: string;
    password: string;
}

export const loginRequestValidator = yup.object({
    email: yup.string().email("Invalid email address").required("Email is required"),
    password: yup.string().required("Password is required")
});

File: VotingSystem.Shared/Models/LoginRequestDto.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace VotingSystem.Shared.Models
{
    public record LoginRequestDto
    {
        /// <summary>
        /// Email of the user
        /// </summary>
        [Required(ErrorMessage = "Email megadása kötelező")]
        [EmailAddress(ErrorMessage = "Helytelen email formátum")]
        public required string Email { get; init; }

        /// <summary>
        /// Password of the user
        /// </summary>
        [Required(ErrorMessage = "Jelszó megadása kötelező")]
        public required string Password { get; init; }
    }

}
